x-pack/plugins/fleet/public/types/ui_extensions.ts

```diff
@@ -20,6 +29,23 @@ export interface UIExtensionsStorage {
   [key: string]: Partial<Record<UIExtensionPoint['view'], UIExtensionPoint>>;
 }
 
+/**
+ * UI Component Extension is used to replace the Define Step on
+ * the pages displaying the ability to edit/create an Integration Policy
+ */
+export type PackagePolicyReplaceDefineStepExtensionComponent =
+  ComponentType<PackagePolicyReplaceDefineStepExtensionComponentProps>;
+
+export type PackagePolicyReplaceDefineStepExtensionComponentProps = (
+  | (PackagePolicyEditExtensionComponentProps & { isEditPage: true })
+  | (PackagePolicyCreateExtensionComponentProps & { isEditPage: false })
+) & {
+  validationResults?: PackagePolicyValidationResults;
+  agentPolicy?: AgentPolicy;
+  packageInfo: PackageInfo;
+  integrationInfo?: RegistryPolicyTemplate;
+};
```

Comment:
since this extension replaces the first step entirely, and it slightly differs between `edit` and `create`, i've used the same props from the relevant existing extension points. 

the added props are from the actual props `StepDefinePackagePolicy` receives, so any functionality it does inside an be replicated. 

`integrationInfo` is added to be able to tell if installation is a top package or a policy template.